build.rs

```diff
@@ -1,17 +1,25 @@
 // Copyright 2018 the Deno authors. All rights reserved. MIT license.
 
 // Run "cargo build -vv" if you want to see gn output.
-// TODO For the time being you must set an env var DENO_BUILD_PATH
-// which might be `pwd`/out/debug or `pwd`/out/release.
-// TODO Currently DENO_BUILD_PATH must be absolute.
 // TODO Combine DENO_BUILD_PATH and OUT_DIR.
 
 use std::env;
+use std::path::Path;
 use std::process::Command;
 
 fn main() {
+  // Cargo sets PROFILE to either "debug" or "release", which conveniently
+  // matches the build modes we support.
   let mode = env::var("PROFILE").unwrap();
-  let deno_build_path = env::var("DENO_BUILD_PATH").unwrap();
+
+  let repo_root_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+  let repo_root_dir = Path::new(&repo_root_dir);
+
+  let deno_build_path = match env::var("DENO_BUILD_PATH") {
+    Ok(s) => repo_root_dir.join(s),
+    Err(_) => repo_root_dir.join("out").join(&mode),
```

Comment:
This is inverting control - because build.rs is running, it means we're a cargo build, but now we're ceding the output directory to how setup.py wants it.  This is brittle because there are two place where "out/$profile" are hardcoded - setup.py and here. 

We're just about to run `tools/setup.py` - why not tell it to gn gen into the directory we want it to. Cargo knows where it wants to put stuff, it has an output directory - that is OUT_DIR - we should not invert control and rather tell gn/ninja where to put build files.

Reply:
We *could* put any path here. 
Since set DENO_BUILD_PATH when calling setup.py, it'll use that path.

However I prefer it to be the same because that allows me to use RLS without building v8 to a separate directory. 

Although I do think it would make sense to make ninja build to OUT_DIR in some scenarios - maybe even the default - this has to be supported one way or another and it *cannot* be configured with an environment variable because I can't control the env vars that vscode passes to rls.

So until we add a config file, let's just land this.

Reply:
I'm ok as landing this but it should be marked with a TODO - this is a hack to work around ccache and/or other problems with your setup - it's not the way it should be.

The desire to not build v8 in different directories is not a valid concern reason to invert build system control. If ccache doesn't work, do something hacky locally, like symlinking the directories.

I'm very concerned about not letting hacks get layered on top of hacks when interfacing with these build systems. We need to think carefully about what is happening - cargo is being run - it has created a directory for us to build into - we need to produce some static libraries so it can link to them. We have all the tools at our disposal to produce those static libraries in the provided OUT_DIR directory.

Reply:
> So until we add a config file, let's just land this.

I want to avoid any config files. We need to reduce the number of variables that control the build systems.

Reply:
Another idea for a hack you can use locally: `cargo check --target-dir out/debug`

Reply:
> Another idea for a hack you can use locally: `cargo check --target-dir out/debug`

Again, I cannot control the invocation flow from vscode -> rls -> cargo.